### ISSUE TITLE
Label is forcibly rendered with WindowText color instead of ControlText in high contrast mode

### ISSUE THREAD

**Issue by koszeggy** [dotnet/winforms#5196]
* .NET Core Version: 
All versions from .NET Core 3.0 to .NET 5

* Have you experienced this same bug with .NET Framework?: 
No

**Problem description:**

The default color of a `Label` is `SystemColors.ControlText`. However, starting with .NET Core 3.0, it is rendered with `SystemColors.WindowText` color in HighContrast mode. This can produce the following issue with a high contrast color set using inverted Window/Control colors:

![High contrast .NET Framework vs Core](https://user-images.githubusercontent.com/27336165/124311994-e5030d00-db6e-11eb-84e4-64671b068504.png)
On the left side there is the .NET Framework 4.5 version, whereas on the right the .NET 5.0. Notice the disappeared labels.

**Expected behavior:**

In high contrast mode, a `Label` with default colors should not disappear on a control that also has the default `SystemColors.Control` color.

**Minimal repro:**
* Create a customized high contrast theme where Control and Window colors are inverses of each other.
* Create a `Form` with a `Label` on it. Or just use [this application](https://github.com/koszeggy/KGySoft.Drawing.Tools), which is the source of the screenshot above.

> _Note:_ It's actually doesn't really make sense to create such a theme because Window/Control colors are often misused even in Windows settings but it is very useful for testing purposes.

**Further remarks:**
The culprit is [this](https://github.com/dotnet/winforms/blob/3fd1247e1c3f0edd2af4dff5114db998f8b96803/src/System.Windows.Forms/src/System/Windows/Forms/Label.cs#L1308) line of code, which is not present in the [framework version](https://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/WinForms/Label.cs,1527). Unfortunately this solution overrides any explicitly set colors, so it cannot be fixed without overriding `OnPaint` or handling the `Paint` event.

Btw, this theme helped me to detect other issues, too; however these can be fixed by explicitly set colors:
* A `TextBox` uses `Window` and `WindowText` colors, but in `ReadOnly` mode the background changes to `Control`, while text color remains `WindowText`. Fortunately `TextBox` allows explicitly set colors so I can adjust the `ForeColor` in `ReadOnly` mode
* Similarly to `Label`, the default cell style of a `DatagridView` also uses ill-paierd colors (and also the default header cell style), but these can be overridden as well.

**Comment by RussKie** [comment]
@vladimir-krestov @SergeySmirnov-Akvelon please investigate

**Comment by Jenna-Zheng** [comment]
Verified this issue with .NET dotnet-sdk-6.0.0-rc.1.21368.1 from latest main branch, it is fixed.
![image](https://user-images.githubusercontent.com/64949616/126137251-f3f2db17-5b14-495c-b783-cf63de6b0ee7.png)

**Comment by Olina-Zhang** [comment]
Verified with the .Net 6.0 Preview 7 release build: 6.0.100-Preview.7.21369.19, it is fixed as above result.
